pkg/RepoManages.go

```diff
@@ -387,15 +403,71 @@ func (impl RepoManagerImpl) FetchChanges(pipelineMaterialId int, from string, to
 		return response, nil
 	}
 	commits := make([]*git.GitCommit, 0)
-	err = json.Unmarshal([]byte(pipelineMaterial.CommitHistory), &commits)
+	err := json.Unmarshal([]byte(pipelineMaterial.CommitHistory), &commits)
 	if err != nil {
 		return nil, err
 	}
 	response.Commits = commits
 	return response, nil
+}
+
+
+func (impl RepoManagerImpl) FetchGitCommitsForPrType(pipelineMaterial *sql.CiPipelineMaterial) (*git.MaterialChangeResp, error) {
```

Comment:
should it be PR or should it be webhook? question is - is this generic enough to handle any webhook for CI trigger or only PR based trigger

Reply:
Done